src/precice/impl/Participant.hpp

```diff
@@ -73,98 +72,178 @@ class Participant {
-   * @return a pointer to the MeshContext or nullptr if it was not found
+   * @return a reference to the MeshContext
+   * @pre there is a matching mesh
    */
-  MeshContext const *usedMeshContextByName(const std::string &name) const;
+  MeshContext const &usedMeshContext(const std::string &name) const;
 
-  void addReadMappingContext(MappingContext *mappingContext);
+  /** Looks for a used MeshContext with a given mesh ID.
+   * @param[in] meshID the id of the \ref Mesh
+   * @return a reference to the MeshContext
+   * @pre there is a matching mesh
+   */
+  MeshContext &usedMeshContext(int meshID);
 
-  void addWriteMappingContext(MappingContext *mappingContext);
+  /** Looks for a used MeshContext with a given meshID
+   * @param[in] meshID the id of the \ref Mesh
+   * @return a reference to the MeshContext
+   * @pre there is a matching mesh
+   */
+  MeshContext const &usedMeshContext(int meshID) const;
 
-  const utils::ptr_vector<MappingContext> &readMappingContexts() const;
+  /// Does preCICE know a mesh with this meshID?
+  bool hasMesh(int meshID) const;
 
-  const utils::ptr_vector<MappingContext> &writeMappingContexts() const;
+  /// Does preCICE know a mesh with this name?
+  bool hasMesh(const std::string &meshName) const;
 
-  void addWatchPoint(const PtrWatchPoint &watchPoint);
+  /// Is a mesh with this id used by this participant?
+  bool isMeshUsed(int meshID) const;
 
-  void addWatchIntegral(const PtrWatchIntegral &watchIntegral);
+  /// Is a mesh with this name used by this participant?
+  bool isMeshUsed(const std::string &meshID) const;
 
-  std::vector<PtrWatchPoint> &watchPoints();
+  /// Is a mesh with this id provided?
+  bool isMeshProvided(int meshID) const;
 
-  std::vector<PtrWatchIntegral> &watchIntegrals();
+  /// Get the used mesh id of a mesh with this name.
+  int getUsedMeshID(const std::string &meshName) const;
 
-  /// Adds a mesh to be used by the participant.
-  void useMesh(
-      const mesh::PtrMesh &                         mesh,
-      const Eigen::VectorXd &                       localOffset,
-      bool                                          remote,
-      const std::string &                           fromParticipant,
-      double                                        safetyFactor,
-      bool                                          provideMesh,
-      partition::ReceivedPartition::GeometricFilter geoFilter);
+  /// Get the name of a mesh given by its id.
+  std::string getMeshName(int meshID) const;
 
-  void addAction(action::PtrAction &&action);
+  /// Get a mesh name which uses the given data id.
+  std::string getMeshNameFromData(int dataID) const;
+  /// @}
 
-  std::vector<action::PtrAction> &actions();
+  /// @name Other querries
+  /// @{
+  /// Returns the name of the participant.
+  const std::string &getName() const;
 
-  const std::vector<action::PtrAction> &actions() const;
+  /// Returns true, if the participant uses a master process.
+  bool useMaster();
```

Comment:
```suggestion
  bool useMaster() const;
```
? And is 'using' here equal to 'runs on'?

Reply:
Documented what this actually mean. Note that I just reordered these declarations.

Reply:
What about `const` ing the function? If I misunderstand it you can just resolve it.

Reply:
Ah got it, may bad